flink-models/flink-model-openai/src/main/java/org/apache/flink/model/openai/AbstractOpenAIModelFunction.java

```diff
@@ -18,117 +18,86 @@
-                    .defaultValue(ContextOverflowAction.TRUNCATED_TAIL)
-                    .withDescription(
-                            Description.builder()
-                                    .text("Action to handle context overflows. Supported actions:")
-                                    .linebreak()
-                                    .text(ContextOverflowAction.getAllValuesAndDescriptions())
-                                    .build());
-
     protected transient OpenAIClientAsync client;
 
+    private final ErrorHandlingStrategy errorHandlingStrategy;
     private final int numRetry;
+    private final ErrorHandlingStrategy retryFallbackStrategy;
     private final String baseUrl;
     private final String apiKey;
     private final String model;
     @Nullable private final Integer maxContextSize;
     private final ContextOverflowAction contextOverflowAction;
+    protected final List<String> outputColumnNames;
 
     public AbstractOpenAIModelFunction(
             ModelProviderFactory.Context factoryContext, ReadableConfig config) {
-        String endpoint = config.get(ENDPOINT);
+        String endpoint = config.get(OpenAIOptions.ENDPOINT);
         this.baseUrl = endpoint.replaceAll(String.format("/%s/*$", getEndpointSuffix()), "");
-        this.apiKey = config.get(API_KEY);
-        // The model service enforces rate-limiting constraints, necessitating retry mechanisms in
-        // most operational scenarios. Within the asynchronous operator framework, the system is
-        // designed to process up to
-        // config.get(ExecutionConfigOptions.TABLE_EXEC_ASYNC_LOOKUP_BUFFER_CAPACITY) concurrent
-        // requests in parallel. To mitigate potential performance degradation from simultaneous
-        // requests, a dynamic retry strategy is implemented where the maximum retry count is
-        // directly proportional to the configured parallelism level, ensuring robust error
-        // resilience while maintaining throughput efficiency.
+        this.apiKey = config.get(OpenAIOptions.API_KEY);
+
+        this.errorHandlingStrategy = config.get(OpenAIOptions.ERROR_HANDLING_STRATEGY);
         this.numRetry =
-                config.get(ExecutionConfigOptions.TABLE_EXEC_ASYNC_LOOKUP_BUFFER_CAPACITY) * 10;
-        this.model = config.get(MODEL);
-        this.maxContextSize = config.get(MAX_CONTEXT_SIZE);
-        this.contextOverflowAction = config.get(CONTEXT_OVERFLOW_ACTION);
+                this.errorHandlingStrategy == ErrorHandlingStrategy.RETRY
+                        ? config.get(OpenAIOptions.RETRY_NUM)
+                        : 0;
+        this.model = config.get(OpenAIOptions.MODEL);
+        this.maxContextSize = config.get(OpenAIOptions.MAX_CONTEXT_SIZE);
+        this.contextOverflowAction = config.get(OpenAIOptions.CONTEXT_OVERFLOW_ACTION);
+        this.retryFallbackStrategy = config.get(OpenAIOptions.RETRY_FALLBACK_STRATEGY);
+        Preconditions.checkArgument(
```

Comment:
Can we introduce an enum class, which only contains the valid options, for `RETRY_FALLBACK_STRATEGY`